stdlib/Random/src/misc.jl

```diff
@@ -191,31 +186,22 @@ optionally supplying the random-number generator `rng`.
 
 # Examples
 ```jldoctest
-julia> shuffle!(Xoshiro(123), Vector(1:10))
-10-element Vector{Int64}:
-  5
-  4
-  2
-  3
-  6
- 10
-  8
-  1
-  9
-  7
+julia> shuffle!(Xoshiro(0), Vector(1:6))
+6-element Vector{Int64}:
+ 5
+ 1
+ 2
+ 6
+ 3
+ 4
 ```
 """
-function shuffle!(r::AbstractRNG, a::AbstractArray)
+function shuffle!(rng::AbstractRNG, a::AbstractArray)
     # keep it consistent with `randperm!` and `randcycle!` if possible
     require_one_based_indexing(a)
-    n = length(a)
-    @assert n <= Int64(2)^52
-    n == 0 && return a
-    mask = 3
-    @inbounds for i = 2:n
-        j = 1 + rand(r, ltm52(i, mask))
+    @inbounds for i = 2:length(a)
```

Comment:
Can we delete this `@inbounds`, given that this is out-of-bounds for OffsetArrays? Benchmarking in #57771 suggests this `@inbounds` has almost no effect on speed.

Reply:
We currently support one one-based indexing, but yes I will re-benchmark without the `@inbounds`.